Classes/PlusOnes.lua

```diff
@@ -2,126 +2,970 @@
+            and not GL:empty(name)
+        ) then
+            if (not DetailsByPlayerName[name]) then
+                GL:tableSet(DetailsByPlayerName, name .. ".total", plusOne);
+            end
+        end
+    end
 
-    playerName = GL:normalizedName(playerName);
+    self.MaterializedData.DetailsByPlayerName = DetailsByPlayerName;
+end
 
-    if (not DB.PlusOnes[playerName]) then
-        DB.PlusOnes[playerName] = 0;
+--- Format a plus one.
+---
+---@param plusOne any
+---@return number if valid, else nil
+function PlusOnes:toPlusOne(plusOne)
+    plusOne = tonumber(plusOne);
+
+    if not plusOne then
+        return nil;
     end
 
-    DB.PlusOnes[playerName] = 0;
+    plusOne = math.max(0, math.floor(plusOne));
+    return plusOne;
+end;
+
+--- Clear all plus one data
+---
+---@return void
+function PlusOnes:clearPlusOnes()
+    DB.PlusOnes = {
+        Totals = {},
+        MetaData = {},
+    };
+    self.MaterializedData = {
+        DetailsByPlayerName = {},
+    };
+    
+    ---added this code below to broadcast when the ML clears all the plus one data.
+    if (GL.Settings:get("PlusOnes.automaticallyShareData")
+    and self:userIsAllowedToBroadcast()) then
+        self:broadcast(); 
+    end
+    ---
+
     self:triggerChangeEvent();
+end
+
+--[[--- Determine if a player is present in the table --- not sure if this is even called elsewhere
+---
+---@param name string
+---@return boolean
+function PlusOnes:hasPlusOnes(name)
+    GL:debug("PlusOnes:hasPlusOnes");
 
-    return 0;
+    if (type(name) ~= "string") then
+        return false;
+    end
+
+    local normalizedName = self:normalizedName(name);
+
+    if (self.MaterializedData.DetailsByPlayerName[normalizedName]) then
+        return true;
+    end
+
+    return false;
+end]]--
+
+--- Get a player's plus one
+---
+---@param name string
+---@return void
+function PlusOnes:getPlusOnes(name)
+    GL:debug("PlusOnes:getPlusOnes");
+
+    if (type(name) ~= "string") then
+        return 0;
+    end
+
+    local normalizedName = self:normalizedName(name);
+
+    return GL:tableGet(self.MaterializedData or {}, "DetailsByPlayerName." .. normalizedName .. ".total", 0);
 end
 
---- Clear all plusOnes
+--- Set a player's plus one
 ---
+---@param name string
+---@param plusOne number
+---@param dontBroadcast boolean
 ---@return void
-function PlusOnes:clear()
-    GL:debug("PlusOnes:clear");
+function PlusOnes:setPlusOnes(name, plusOne, dontBroadcast)
+    GL:debug("PlusOnes:setPlusOnes");
+    if (type(name) ~= "string") then
+        return;
+    end
 
-    DB.PlusOnes = {};
+    dontBroadcast = GL:toboolean(dontBroadcast);
 
+    local normalizedName = self:normalizedName(name);
+
+    if (not self.MaterializedData.DetailsByPlayerName[normalizedName]) then
+        GL:tableSet(self.MaterializedData or {}, "DetailsByPlayerName." .. normalizedName .. ".total", plusOne);
+    else
+        self.MaterializedData.DetailsByPlayerName[normalizedName].total = plusOne;
```

Comment:
If `self.MaterializedData` is not available in the `tableSet` above, then it will also fail here, because it still won't be set at this point

Reply:
I'm pretty sure I copied this code almost verbatim from the boosted rolls class.  I will check and see if that's the case because both instances will need to be fixed .